lib/libzfs/libzfs_pool.c

```diff
@@ -2482,7 +2519,7 @@ zpool_vdev_offline(zpool_handle_t *zhp, const char *path, boolean_t istmp)
 		return (zfs_error(hdl, EZFS_ISSPARE, msg));
 
 	zc.zc_cookie = VDEV_STATE_OFFLINE;
-	zc.zc_obj = istmp ? ZFS_OFFLINE_TEMPORARY : 0;
+	zc.zc_obj = aux;
```

Comment:
It looks like `zc.zc_obj` is used differently depending on the state.

When `zc.zc_cookie=VDEV_STATE_ONLINE|VDEV_STATE_OFFLINE` then `zc.zc_obj` is treated as one of the `ZFS_IOC_VDEV_SET_STATE` flags.  But when `zc.zc_cookie=VDEV_STATE_FAULTED|VDEV_STATE_DEGRADED` then it's expected to be a `vdev_aux_t`.

Using `ZFS_OFFLINE_TEMPORARY` here is right and what the kernel expects.  We don't want to give the impression an arbitrary `vdev_aux_t` can be passed.  We should stick with the original code.